### PR TITLE
[next] Auth Popup v2

### PR DESCRIPTION
**Pull request by alinhle**
## What does this PR do?
- https://www.pivotaltracker.com/story/show/161655175
- New design + Social Login

### CODE REVIEW

File: scripts/start.ts [hunk 1]
```diff
@@ -52,45 +51,30 @@ if (process.env.HOST) {
...
-    const devServer = new WebpackDevServer(compiler, serverConfig);
-    // Launch WebpackDevServer.
-    devServer.listen(port, HOST, (err: Error) => {
-      if (err) {
-        return console.log(err);
-      }
-      console.log(chalk.cyan("Starting the development server...\n"));
-    });
+const protocol = "http";
+const appName = "Talk";
```
Comment (wyattjoh) [inline scripts/start.ts]: As these are "thankfully" rarely to change, I'd suggest just in-lining these variables instead.

File: src/core/client/auth/local/initLocalState.ts [hunk 2]
```diff
@@ -1,28 +1,33 @@
...
  * Initializes the local state, before we start the App.
  */
-export default async function initLocalState(environment: Environment) {
-  commitLocalUpdate(environment, s => {
-    const root = s.getRoot();
+export default async function initLocalState(
+  environment: Environment,
+  context: TalkContext
+) {
+  const authToken = window.location.hash
```
Comment (wyattjoh) [inline src/core/client/auth/local/initLocalState.ts:8]: Why not extract this to a separate function?

Comment (alinhle) [reply src/core/client/auth/local/initLocalState.ts:8]: Sure

File: scripts/start.ts (hunk 1 above)
Comment (alinhle) [reply scripts/start.ts]: Sure

**Review by wyattjoh — APPROVED** [review]

**Pull request merged**